### PR TITLE
rbd-mirror: ignore errors relating to parsing the cluster config file

### PR DESCRIPTION
**Pull request by dillaman**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/common/ConfUtils.cc [hunk 1]
```diff
@@ -122,7 +122,8 @@ int ConfFile::parse_file(const std::string &fname,
       return -EINVAL;
     }
   } catch (const fs::filesystem_error& e) {
-    if (fs::is_other(fname)) {
+    std::error_code ec;
+    if (fs::is_other(fname, ec) && !ec) {
```
Comment (tchaikov) [inline src/common/ConfUtils.cc]: i think we need to check ec first, as, if ec is non-zero, the return value won't be a valid one anymore. so something like
```c++
if (auto is_other = fs::is_other(fname, ec); !ec && is_other) {
```

File: src/global/global_init.cc [hunk 2]
```diff
@@ -214,21 +214,32 @@ global_init(const std::map<std::string,std::string> *defaults,
...
+      if (!uid && !p) {
+	cerr << "unable to look up user '" << g_conf()->setuser << "'"
+	     << std::endl;
+	exit(1);
+      }
+
+      if (p) {
 	uid = p->pw_uid;
 	gid = p->pw_gid;
 	uid_string = g_conf()->setuser;
```
Comment (trociny) [inline src/global/global_init.cc]: Nit: I suppose now, if 'setuser' option is specified as numeric (uid) value, `uid_string` will be set to this value, while it is expected to contain the user name. The previous behavior in this case seemed to have `uid_string` empty. I am not sure what is better (as it seems to be used only for diagnostic), but I suppose we could improve this by using `p->pw_name` here?

File: src/common/ConfUtils.cc (hunk 1 above)
Comment (dillaman) [reply src/common/ConfUtils.cc]: If there is an error, `is_other` returns false (the `file_status` is set to 'unknown'), so it falls through. The `ec` was mostly just to stop the exception from being raised.

File: src/global/global_init.cc (hunk 2 above)
Comment (dillaman) [reply src/global/global_init.cc]: I tweaked it to resemble the old behaviour as closely as possible.

File: src/common/ConfUtils.cc (hunk 1 above)
Comment (tchaikov) [reply src/common/ConfUtils.cc]: > If there is an error, `is_other` returns false

ahh, right!

**Pull request merged**